### DISCUSSION TITLE
How to solve this multi dimensional optimization problem?

### DISCUSSION THREAD

**Discussion by ngaihiang** [google/or-tools#3152] [Q&A]
I have an optimisation problem but I am not sure how to model it 

Optimization Problem: 
I have multiple machines, and each machine has 2 cartridges. 

Assuming I have 10 machines and 25 cartridges, how should I allocate it over a period of 12 months? 

I have constraints such as some cartridge can only serve for 1/3/4/x months.. lets call this its remaining useful life (RUL). Each month that a cartridge serves, its RUL decreases by one. 

Some cartridge once its past its RUL, will have to be sent for checks and be out of service for 1 month.  When that happens, we will swap another available cartridge into the machine.
Once the cartridge is back from checks, it can go on to serve the machine for another 12 months again before it needs to be go back for checks. 

So here we have a time dimension over 12 months of planning, while working with machines and cartridges... How can I go about modelling this problem since there seems to be 3 dimensions to it. 

My objective function would be to reduce the number of times that we have to swap out the cartridges because most of my costs comes from this swap..

Should this be formulated as an assignment or scheduling or travelling salesman problem? Also, there is the problem where existing machines already have cartridges mounted on it.. so there is kind of a "starting point" to consider as well..

Appreciate any help and thanks in advance!

**Comment by lperron** [comment]
I do not see the problem. You swap cards every 12 months. Where do you think you will gain money ?
